src/array.cr

```diff
@@ -2029,6 +2029,31 @@ class Array(T)
     @capacity - @offset_to_buffer
   end
 
+  private def rewind
+    root_buffer.copy_from(@buffer, @size)
+    shift_buffer_by -@offset_to_buffer
+  end
+
+  # Enusures that the internal buffer has at least `capacity` elements.
+  def ensure_capacity(capacity : Int32) : self
+    rewind
+
+    if capacity >= @size
+      resize_to_capacity capacity
+    end
+    self
+  end
+
+  # Reduces the internal buffer to exactly fit the number of elements in the
+  # array, plus `extra` elements.
+  def trim_to_size(extra : Int32 = 0) : self
+    raise ArgumentError.new("Negative extra capacity: #{extra}") if extra < 0
+
+    rewind
+    resize_to_capacity (@size + extra)
```

Comment:
Hmm there's an extra space here; surprised `crystal tool format` didn't complain

Reply:
Interesting. I suppose that's because it's just interpreted as method call plus a separate parenthesized expression, not as a method call with args in parenthesis.